pkg/apis/ark/v1/backup_locations.go

```diff
@@ -0,0 +1,89 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupStorageLocation is a location where Ark stores backup objects.
+type BackupStorageLocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   BackupStorageLocationSpec   `json:"spec"`
+	Status BackupStorageLocationStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupStorageLocationList is a list of BackupStorageLocations.
+type BackupStorageLocationList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Items             []BackupStorageLocation `json:"items"`
+}
+
+// ObjectStorageLocation specifies the settings necessary to connect to a provider's object storage.
+type ObjectStorageLocation struct {
+	// Provider is the cloud provider where object storage resides.
+	Provider string `json:"provider"`
+
+	// Config is for provider-specific configuration fields.
+	Config map[string]string `json:"config"`
+
+	// Bucket is the bucket to use for object storage.
+	Bucket string `json:"bucket"`
+
+	// Prefix is the path inside a bucket to use for Ark storage. Optional.
+	Prefix string `json:"prefix"`
+}
+
+// BackupStorageLocationSpec defines the specification for an Ark BackupStorageLocation
+type BackupStorageLocationSpec struct {
+	ObjectStorage ObjectStorageLocation `json:"objectStorage"`
+}
+
+// BackupStorageLocationPhase is a string representation of the lifecyle phase
+// of an Ark BackupStorageLocation
+type BackupStorageLocationPhase string
+
+const (
+	// BackupStorageLocationPhaseAvailable means the location is available to read and write from.
+	BackupStorageLocationPhaseAvailable BackupStorageLocationPhase = "Available"
+
+	// BackupStorageLocationPhaseUnavailable means the location is unavailable to read and write from.
+	BackupStorageLocationPhaseUnavailable BackupStorageLocationPhase = "Unavailable"
+)
+
+// BackupStorageLocationAccessMode is a string representation of the permissions
+// for a BackupStorageLocation.
+type BackupStorageLocationAccessMode string
+
+const (
+	// BackupStorageLocationAccessModeReadOnly represents read-only access to a BackupStorageLocation.
+	BackupStorageLocationAccessModeReadOnly BackupStorageLocationAccessMode = "ReadOnly"
+
+	// BackupStorageLocationAccessModeReadWrite represents read and write access to a BackupStorageLocation.
+	BackupStorageLocationAccessModeReadWrite BackupStorageLocationAccessMode = "ReadWrite"
+)
+
+// BackupStorageLocationStatus describes the current status of an Ark BackupStorageLocation.
+type BackupStorageLocationStatus struct {
+	Phase      BackupStorageLocationPhase      `json:"phase"`
```

Comment:
Let's add an `omitempty` tag to each of these status fields. I'm not sure we'll actually get around to actually implementing anything that uses them until later, so will be nice not to actually show them in json/yaml.